microprofile/opentracing-extension/src/main/java/org/wildfly/extension/microprofile/opentracing/TracerAttributes.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2019 JBoss by Red Hat.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.extension.microprofile.opentracing;
+
+
+import org.jboss.as.controller.PropertiesAttributeDefinition;
+import org.jboss.as.controller.SimpleAttributeDefinition;
+import org.jboss.as.controller.SimpleAttributeDefinitionBuilder;
+import org.jboss.as.controller.StringListAttributeDefinition;
+import org.jboss.as.controller.capability.RuntimeCapability;
+import org.jboss.as.controller.client.helpers.MeasurementUnit;
+import org.jboss.as.network.OutboundSocketBinding;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
+import org.wildfly.microprofile.opentracing.smallrye.TracerConfigurationConstants;
+
+/**
+ * Tracer attribute definitions.
+ *
+ * @author Emmanuel Hugonnet (c) 2019 Red Hat, Inc.
+ */
+public class TracerAttributes {
+
+    private static final String OUTBOUND_SOCKET_BINDING_CAPABILITY_NAME = "org.wildfly.network.outbound-socket-binding";
+    static final RuntimeCapability<Void> OUTBOUND_SOCKET_BINDING_CAPABILITY =
+            RuntimeCapability.Builder.of(OUTBOUND_SOCKET_BINDING_CAPABILITY_NAME,
+                    true, OutboundSocketBinding.class)
+                    .build();
+
+    public static final StringListAttributeDefinition PROPAGATION = StringListAttributeDefinition.Builder.of(TracerConfigurationConstants.PROPAGATION)
+            .setAllowNullElement(false)
+            .setRequired(false)
+            .setAllowedValues("JAEGER", "B3")
+            .setAttributeGroup("codec-configuration")
+            .setRestartAllServices()
```

Comment:
Why must this require a restart of all services?  Same question for all attributes below.

Reply:
Because this is a configuration of a tracer, not a tracer itself thus you need to recreate a Tracer instance to update it which means restarting the deployments that have the tracer already injected.

Reply:
But we would only need to restart those deployments with a dependency on the specific tracer configuration, not all deployments, right?
In that case you can use setFlags(AttributeAccess.Flag.RESTART_RESOURCE_SERVICES).

Reply:
ok

Reply:
AttributeAccess.Flag.RESTART_RESOURCE_SERVICES is something of an anti-pattern as a typical user won't understand that bouncing a service means bouncing the deployment. And bouncing a deployment isn't graceful.

This seems like a possible future enhancement.

Reply:
ok so keeping my restartAllServices